packages/block-library/src/navigation/edit/edit-menu-toolbar-button.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockControls,
+} from '@wordpress/block-editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+import {
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
+import { store as interfaceStore } from '@wordpress/interface';
+import { __ } from '@wordpress/i18n';
+
+const EditMenuToolbarButton = () => {
+ 	// Blocks can be loaded into both post and site editors.
+	// We use this to determine which editor we are in so that
+	// we can determine which inspector controls to open.
+	const {
+		isPostEditor,
+	} = useSelect(
+		( select ) => {
+			// eslint-disable-next-line @wordpress/data-no-store-string-literals
+			const editorSelectors = select( 'core/editor' );
+			return {
+				isPostEditor: !! editorSelectors.getEditedPostAttribute( 'type' )
+			};
+		}
+	);
+	const { enableComplementaryArea } = useDispatch( interfaceStore );
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
```

Comment:
What happens if it's neither edit-post nor edit-post. How should the navigation block behave?